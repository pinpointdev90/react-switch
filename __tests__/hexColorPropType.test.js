import hexColorPropType from "../src/hexColorPropType";

describe("hexColorPropType()", () => {
  it("returns null when passed correct color", () => {
    expect(
      hexColorPropType({ onColor: "#abcdef" }, "onColor", "ReactSwitch")
    ).toBeNull();
  });
  it("returns error when passed plain text color", () => {
    expect(
      hexColorPropType({ offColor: "blue" }, "offColor", "ReactSwitch")
    ).toMatchSnapshot();
  });
  it("returns error when passed non string", () => {
    expect(
      hexColorPropType({ onHandleColor: ["#", "0", "0", "0"] })
    ).toMatchSnapshot();
  });
  it("returns error when passed 8-digit hex-color", () => {
    expect(
      hexColorPropType(
        { offHandleColor: "#11223344" },
        "offHandleColor",
        "ReactSwitch"
      )
    ).toMatchSnapshot();
  });
});
