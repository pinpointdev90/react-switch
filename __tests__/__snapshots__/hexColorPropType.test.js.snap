// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hexColorPropType() returns error when passed 8-digit hex-color 1`] = `[Error: Invalid prop 'offHandleColor' supplied to 'ReactSwitch'. 'offHandleColor' has to be either a 3-digit or 6-digit hex-color string. Valid examples: '#abc', '#123456']`;

exports[`hexColorPropType() returns error when passed non string 1`] = `[Error: Invalid prop 'undefined' supplied to 'undefined'. 'undefined' has to be either a 3-digit or 6-digit hex-color string. Valid examples: '#abc', '#123456']`;

exports[`hexColorPropType() returns error when passed plain text color 1`] = `[Error: Invalid prop 'offColor' supplied to 'ReactSwitch'. 'offColor' has to be either a 3-digit or 6-digit hex-color string. Valid examples: '#abc', '#123456']`;
